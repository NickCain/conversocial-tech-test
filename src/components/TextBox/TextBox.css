.Textbox {
    text-align: left;
}

.Textbox textarea {
    width: 100%;
}
