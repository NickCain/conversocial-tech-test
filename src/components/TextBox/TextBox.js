import React, { Component } from 'react';

import './TextBox.css';

class TextBox extends Component {
  render() {
    return (
      <div className="Textbox">
        <textarea placeholder="write your message here..."></textarea>
      </div>
    );
  }
}

export default TextBox;
