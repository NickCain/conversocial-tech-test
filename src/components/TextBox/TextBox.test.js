import { mount, shallow } from 'enzyme';
import React from 'react';
import TextBox from './TextBox.js'

let wrapper;
let shallowWrapper;

describe('TextBox - component', () => {
  describe('TextBox renders', () => {
    beforeEach(() => {
      wrapper = mount(
        <TextBox />
      );
      shallowWrapper = shallow(
        <TextBox />
      );
    });

    it('should render the TextBox component', () => {
      expect(shallowWrapper).toMatchSnapshot();
      expect(wrapper.length).toBeGreaterThan(0);
    });
  });
});
