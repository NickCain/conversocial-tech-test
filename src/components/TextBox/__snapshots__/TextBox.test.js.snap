// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox - {type} TextBox renders should render the TextBox component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="Textbox"
>
    <textarea
        placeholder="write your message here..."
    />
</div>,
  "nodes": Array [
    <div
      className="Textbox"
>
      <textarea
            placeholder="write your message here..."
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TextBox />,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TextBox {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="Textbox"
>
          <textarea
                    placeholder="write your message here..."
          />
</div>,
        "_debugID": 6,
        "_renderedOutput": <div
          className="Textbox"
>
          <textarea
                    placeholder="write your message here..."
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TextBox />,
}
`;

exports[`TextBox - component TextBox renders should render the TextBox component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="Textbox"
>
    <textarea
        placeholder="write your message here..."
    />
</div>,
  "nodes": Array [
    <div
      className="Textbox"
>
      <textarea
            placeholder="write your message here..."
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TextBox />,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TextBox {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="Textbox"
>
          <textarea
                    placeholder="write your message here..."
          />
</div>,
        "_debugID": 6,
        "_renderedOutput": <div
          className="Textbox"
>
          <textarea
                    placeholder="write your message here..."
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TextBox />,
}
`;
